File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_1.py
A = [0] * 10
print(A)
V = [False] * 10
print(V)
def go(index, n, m):
    if index == m:
        for i in range(m):
            print(A[i], end = " ")
        print()
        return
    for i in range(1, n + 1):
        if V[i]:
            continue
        V[i] = True
        A[index] = i
        go(index + 1, n, m)
        V[i] = False


N, M = input().split()
go(0, int(N), int(M))

File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_1.cpp
//중복 불가 수열
#include <iostream>
using namespace std;
bool V[10];
int A[10];
int N, M;

void go(int index, int n, int m) {
	if (index == m) {
		for (int i = 0; i < m; i++) {
			cout << A[i] << " ";
		}
		cout << endl;
		return;
	}
	for (int i = 1; i <= n; i++) {
		if (V[i]) continue;
		V[i] = true; A[index] = i;
		go(index + 1, n, m);
		V[i] = false;
	} 
}

int main() {
	cin >> N >> M;
	go(0, N, M);













	return 0;
}